stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt

```diff
@@ -394,7 +413,15 @@ internal class ChannelLogic(
         repos.insertMessages(messages)
     }
 
-    internal fun upsertMessage(message: Message) = upsertMessages(listOf(message))
+    internal fun upsertMessage(message: Message) {
+        /* Do not upsert message if the newest messages are not loaded to avoid breaking pagination. If the message
+        * is inside the list and was updated in some kind of way, update it in the list. */
+        // if (mutableState._endOfNewerMessages.value ||
+        //     mutableState._messages.value.containsKey(message.id)
```

Comment:
should we remove the commented out code?

Reply:
no we should not, this should have been reverted